subgraphs/_reference_/src/sdk/protocols/bridge/protocol.ts

```diff
@@ -0,0 +1,457 @@
+   * @private
+   */
+  private constructor(
+    protocol: BridgeProtocolSchema,
+    pricer: TokenPricer,
+    event: ethereum.Event
+  ) {
+    this.protocol = protocol;
+    this.event = event;
+    this.pricer = pricer;
+    this.snapshoter = new ProtocolSnapshot(protocol, event);
+  }
+
+  /**
+   * This is the main function to instantiate a Bridge entity. Most times it is not called directly, but from the SDK initializer.
+   *
+   * @param conf {BridgeConfigurer} An object that implements the BridgeConfigurer interface, to set some of the protocol's properties
+   * @param pricer {TokenPricer} An object that implements the TokenPricer interface, to allow the wrapper to access pricing data
+   * @param event {ethereum.Event} The event being handled at a time.
+   * @returns Bridge
+   */
+  static load(
+    conf: BridgeConfigurer,
+    pricer: TokenPricer,
+    event: ethereum.Event
+  ): Bridge {
+    const id = Address.fromString(conf.getID());
+    let protocol = BridgeProtocolSchema.load(id);
+    if (protocol) {
+      const proto = new Bridge(protocol, pricer, event);
+      proto.setVersions(conf.getVersions());
+      return proto;
+    }
+
+    protocol = new BridgeProtocolSchema(id);
+    protocol.name = conf.getName();
+    protocol.slug = conf.getSlug();
+    protocol.network = dataSource.network().toUpperCase();
+    protocol.type = constants.ProtocolType.BRIDGE;
+    protocol.permissionType = conf.getPermissionType();
+    protocol.totalValueLockedUSD = constants.BIGDECIMAL_ZERO;
+    protocol.protocolControlledValueUSD = constants.BIGDECIMAL_ZERO;
+    protocol.cumulativeSupplySideRevenueUSD = constants.BIGDECIMAL_ZERO;
+    protocol.cumulativeProtocolSideRevenueUSD = constants.BIGDECIMAL_ZERO;
+    protocol.cumulativeTotalRevenueUSD = constants.BIGDECIMAL_ZERO;
+    protocol.cumulativeVolumeInUSD = constants.BIGDECIMAL_ZERO;
+    protocol.cumulativeVolumeOutUSD = constants.BIGDECIMAL_ZERO;
+    protocol.cumulativeTotalVolumeUSD = constants.BIGDECIMAL_ZERO;
+    protocol.netVolumeUSD = constants.BIGDECIMAL_ZERO;
+    protocol.cumulativeUniqueUsers = 0;
+    protocol.cumulativeUniqueTransferSenders = 0;
+    protocol.cumulativeUniqueTransferReceivers = 0;
+    protocol.cumulativeUniqueLiquidityProviders = 0;
+    protocol.cumulativeUniqueMessageSenders = 0;
+    protocol.cumulativeTransactionCount = 0;
+    protocol.cumulativeTransferOutCount = 0;
+    protocol.cumulativeTransferInCount = 0;
+    protocol.cumulativeLiquidityDepositCount = 0;
+    protocol.cumulativeLiquidityWithdrawCount = 0;
+    protocol.cumulativeMessageSentCount = 0;
+    protocol.cumulativeMessageReceivedCount = 0;
+    protocol.supportedNetworks = [];
+    protocol.totalPoolCount = 0;
+    protocol.totalPoolRouteCount = 0;
+    protocol.totalCanonicalRouteCount = 0;
+    protocol.totalWrappedRouteCount = 0;
+    protocol.totalSupportedTokenCount = 0;
+
+    const proto = new Bridge(protocol, pricer, event);
```

Comment:
why not get this at the beginning of the function so you don't need to load it 2x

Reply:
I think I still need it 2x, unless I put all the protocol default values inside the `if`?

Reply:
I see, resolved